### PR TITLE
Add slots to graph

### PR DESCRIPTION
**Pull request by mtsr**
I've added slots to the graph. Your dot code was clean enough that I didn't want to add a dependency for that. But the itertools thing is really useful to have. Take a look and let me know what you think.

### CODE REVIEW

**Comment by mtsr** [comment]
This is what it looks like now:

render_graph dot 3

File: src/render_graph.rs [hunk 1]
```diff
@@ -1,18 +1,37 @@
...
 use crate::{
     dot::{font_tag, DotGraph},
     utils,
 };
 use bevy::render::render_graph::{Edge, NodeId, RenderGraph};
+use itertools::{EitherOrBoth, Itertools};
+
+/// Escape tags in such a way that it is suitable for inclusion in a
+/// Graphviz HTML label.
+pub fn escape_html<'a, S>(s: S) -> Cow<'a, str>
```
Comment (jakobhellermann) [inline src/render_graph.rs:12]: There is already an `escape_html` function in `src/dot.rs`, you can use (and extend) that

File: src/render_graph.rs [hunk 2]
```diff
@@ -21,22 +40,53 @@ pub fn render_graph_dot(graph: &RenderGraph) -> String {
...
-            false => format!("<BR/> {}", font_tag(&format!("-> {}", outputs), "blue", 10)),
-            true => "".into(),
-        };
 
         let label = format!(
-            "<{}<BR />{} {}>",
-            name,
-            font_tag(&type_name, "red", 10),
-            outputs,
+            "<<TABLE><TR><TD PORT=\"title\" BORDER=\"0\" COLSPAN=\"2\">{}<BR/>{}</TD></TR>{}</TABLE>>",
```
Comment (jakobhellermann) [inline src/render_graph.rs:86]:
```suggestion
            "<<TABLE STYLE=\"rounded\"><TR><TD PORT=\"title\" BORDER=\"0\" COLSPAN=\"2\">{}<BR/>{}</TD></TR>{}</TABLE>>",
```
I personally think the rounded edges make the graph more pleasing to look at.

File: src/render_graph.rs [hunk 3]
```diff
@@ -49,23 +99,30 @@ pub fn render_graph_dot(graph: &RenderGraph) -> String {
...
                     input_node,
                     output_node,
                 } => {
-                    dot.add_edge(&node_id(output_node), &node_id(input_node), &[]);
+                    dot.add_edge(
+                        &node_id(output_node),
+                        Some("title:e"),
+                        &node_id(input_node),
+                        Some("title:w"),
+                        &[("style", "dashed")],
```
Comment (jakobhellermann) [inline src/render_graph.rs:124]:
```suggestion
                        &[],
```
Same as above, just a personal preference

**Review by jakobhellermann — APPROVED** [review]
Thank you very much. I've suggested a few minor tweaks, but otherwise this is good to merge.

**Comment by mtsr** [comment]
Made your proposed changes. This is how it looks now:

render_graph dot 3

File: src/render_graph.rs (hunk 3 above)
Comment (mtsr) [reply src/render_graph.rs]: Done.

File: src/render_graph.rs (hunk 2 above)
Comment (mtsr) [reply src/render_graph.rs]: Done.

File: src/render_graph.rs (hunk 1 above)
Comment (mtsr) [reply src/render_graph.rs]: Done. I added escaping for `&` and `\"` as I already ran into issues with it.

**Comment by mtsr** [comment]
Note the minor layout issues with the text flowing outside the nodes. I found this issue https://github.com/rust-lang/rust/pull/76500 . If I understand correctly the graphviz engine doesn't have Roboto so is failing to set the width correctly in the SVG, which is then rendered like this (by GIMP in my case).

The proposed solution is listing multiple fonts.

**Comment by jakobhellermann** [comment]
Thanks! Can you try if `fontname="Roboto, Serif"` or something like that works?

**Comment by mtsr** [comment]
Doesn't seem to work. Not sure, but can't find good documentation about it, either.

**Pull request merged**